d2core/d2ui/textbox.go

```diff
@@ -77,7 +86,21 @@ func (v *TextBox) OnKeyChars(event d2interface.KeyCharsEvent) bool {
 	newText := string(event.Chars())
 
 	if len(newText) > 0 {
-		v.text += newText
+		if v.isNumberOnly {
+			number, err := strconv.Atoi(v.text + newText)
+			if err != nil {
+				v.Debugf("Unable to convert string %s to intager: %s", v.text+newText, err)
+				return false
+			}
+
+			if number <= v.maxValue {
```

Comment:
for this whole block of code, i would suggest inverting some of these if statements to reduce the nesting.
something like this:
```golang
func (v *TextBox) OnKeyChars(event d2interface.KeyCharsEvent) bool {
	if !v.isFocused || !v.visible || !v.enabled {
		return false
	}

	newText := string(event.Chars())
	if !(len(newText) > 0) {
		return false
	}
	
	if !v.isNumberOnly {
		v.text += newText
		return true
	}

	number, err := strconv.Atoi(v.text + newText)
	if err != nil {
		v.Debugf("Unable to convert string %s to intager: %s", v.text+newText, err)
		return false
	}

	if number <= v.maxValue {
		v.text += newText
	} else {
		v.text = strconv.Itoa(v.maxValue)
	}
	
	return true
}
```

Reply:
oh, also, `integer` instead of `intager` in the debug print.